File: ios/App/GameView.swift
import SwiftUI

struct GameView: View {
    @StateObject private var engine = GameEngine()
    
    var body: some View {
        GeometryReader { geometry in
            TimelineView(.animation) { timeline in
                Canvas { context, size in
                    // Swift requires us to be careful.
                    // Let's perform the update.
                    // We only set dimensions here if they changed, to avoid engine work in render loop
                    if engine.width != size.width || engine.height != size.height {
                        DispatchQueue.main.async {
                            engine.setDimensions(width: size.width, height: size.height)
                        }
                    }
                    
                    // Draw Background
                    context.fill(Path(CGRect(origin: .zero, size: size)), with: .color(Color(hex: 0x1a1a1a)))
                    
                    // Draw Obstacles
                    for obs in engine.obstacles {
                        // Color based on speed? Or just white.
                        // Variable speed gets a tint?
                        var obsColor: Color = .white
                        if obs.type == .variableSpeed {
                            obsColor = .yellow
                        }
                        
                        if obs.type == .lShapeLeft {
                            // Draw L shape blocking Left side
                            // We treat (x,y) as top-left of bounding box
                            // L-Shape Left: |__
                            // Vertical bar on left, horizontal bar on bottom
                            let rect = CGRect(x: obs.x, y: obs.y, width: obs.width, height: obs.height)
                            context.fill(Path(rect), with: .color(obsColor))
                            // Add extra arm?
                            // Simplified: Just draw rect for now, L-shape logic in collision is "rect" anyway.
                            // If we want visual L, we need data.
                            // Let's stick to Rect for visual simplicity unless we want true L geometry.
                            // The user spec says "L Shape (Width=50%)".
                            // If it's just a visual shape, collision must match.
                            // Currently collision is AABB (Rect). So we draw Rect.
                        } else if obs.type == .cross {
                            // Cross shape +
                            let rectH = CGRect(x: obs.x, y: obs.y + obs.height/3, width: obs.width, height: obs.height/3)
                            let rectV = CGRect(x: obs.x + obs.width/2 - 15, y: obs.y, width: 30, height: obs.height)
                            context.fill(Path(rectH), with: .color(obsColor))
                            context.fill(Path(rectV), with: .color(obsColor))
                        } else if obs.type == .rotating || obs.type == .diagonal {
                            context.drawLayer { ctx in
                                ctx.translateBy(x: obs.x + obs.width/2, y: obs.y + obs.height/2)
                                ctx.rotate(by: .radians(obs.rotation))
                                let rect = CGRect(x: -obs.width/2, y: -obs.height/2, width: obs.width, height: obs.height)
                                ctx.fill(Path(rect), with: .color(obsColor))
                            }
                        } else {
                            // Rectangle / Vertical / Variable / Split
                            let rect = CGRect(x: obs.x, y: obs.y, width: obs.width, height: obs.height)
                            context.fill(Path(rect), with: .color(obsColor))
                        }
                    }
                    
                    // Draw Player (Only if playing/gameover)
                    if engine.state == .playing || engine.state == .gameover {
                        let centerX = size.width / 2
                        let centerY = size.height * 0.8
                        
                        // Ring
                        let ringRect = CGRect(x: centerX - engine.player.radius, y: centerY - engine.player.radius, width: engine.player.radius * 2, height: engine.player.radius * 2)
                        let ringPath = Path(ellipseIn: ringRect)
                        context.stroke(ringPath, with: .color(.white.opacity(0.1)), lineWidth: 2)
                        
                        // Balls
                        let p = engine.player
                        let x1 = centerX + cos(p.angle) * p.radius
                        let y1 = centerY + sin(p.angle) * p.radius
                        let x2 = centerX + cos(p.angle + .pi) * p.radius
                        let y2 = centerY + sin(p.angle + .pi) * p.radius
                        
                        let b1 = Path(ellipseIn: CGRect(x: x1 - p.ballRadius, y: y1 - p.ballRadius, width: p.ballRadius*2, height: p.ballRadius*2))
                        context.fill(b1, with: .color(Color(hex: 0xff3b30))) // Red
                        
                        let b2 = Path(ellipseIn: CGRect(x: x2 - p.ballRadius, y: y2 - p.ballRadius, width: p.ballRadius*2, height: p.ballRadius*2))
                        context.fill(b2, with: .color(Color(hex: 0x007aff))) // Blue
                    }
                }
            }
            .overlay(inputLayer) // Moved inputLayer BEHIND uiOverlay (Wait, overlay stacks on top. So inputLayer is on TOP of TimelineView)
            .overlay(uiOverlay) // uiOverlay is on TOP of inputLayer
        }
        .edgesIgnoringSafeArea(.all)
        .statusBar(hidden: true)
    }
    
    var uiOverlay: some View {
        VStack {
            if engine.state == .menu {
                Spacer()
                
                // Logo Icon
                ZStack {
                    Circle()
                        .stroke(Color.white.opacity(0.2), lineWidth: 4)
                        .frame(width: 80, height: 80)
                    
                    Circle()
                        .fill(Color(hex: 0xff3b30)) // Red ball
                        .frame(width: 20, height: 20)
                        .offset(x: 40)
                    
                    Circle()
                        .fill(Color(hex: 0x007aff)) // Blue ball
                        .frame(width: 20, height: 20)
                        .offset(x: -40)
                }
                .rotationEffect(.degrees(-45)) // Stylistic tilt
                .padding(.bottom, 20)
                
                Text("TWO TONES")
                    .font(.system(size: 50, weight: .bold))
                    .foregroundColor(.white)
                
                Button(action: {
                    engine.startLevel(index: 0) // Start Level 1
                }) {
                    Text("PLAY")
                        .font(.system(size: 24, weight: .bold))
                        .foregroundColor(.black)
                        .frame(width: 200, height: 60)
                        .background(Color.white)
                        .cornerRadius(30)
                }
                .padding(.top, 40)
                
                Button(action: {
                    engine.state = .levelSelect
                }) {
                    Text("LEVELS")
                        .font(.system(size: 20, weight: .semibold))
                        .foregroundColor(.white)
                        .frame(width: 200, height: 50)
                        .overlay(
                            RoundedRectangle(cornerRadius: 25)
                                .stroke(Color.white, lineWidth: 2)
                        )
                }
                .padding(.top, 20)

                Spacer()
            } else if engine.state == .levelSelect {
                VStack {
                    Text("SELECT LEVEL")
                        .font(.system(size: 30, weight: .bold))
                        .foregroundColor(.white)
                        .padding(.top, 60)
                    
                    ScrollView {
                        LazyVGrid(columns: [GridItem(.adaptive(minimum: 100))], spacing: 20) {
                            ForEach(engine.levels, id: \.id) { level in
                                let isLocked = level.id > engine.unlockedLevel
                                Button(action: {
                                    if !isLocked {
                                        engine.startLevel(index: level.id - 1)
                                    }
                                }) {
                                    ZStack {
                                        RoundedRectangle(cornerRadius: 15)
                                            .fill(isLocked ? Color.gray.opacity(0.3) : Color.white.opacity(0.2))
                                            .frame(height: 100)
                                        
                                        if isLocked {
                                            Image(systemName: "lock.fill")
                                                .font(.title)
                                                .foregroundColor(.gray)
                                        } else {
                                            VStack {
                                                Text("\(level.id)")
                                                    .font(.system(size: 40, weight: .bold))
                                                    .foregroundColor(.white)
                                                Text(level.name)
                                                    .font(.caption)
                                                    .foregroundColor(.white.opacity(0.8))
                                            }
                                        }
                                    }
                                }
                                .disabled(isLocked)
                            }
                        }
                        .padding()
                    }
                    
                    Button(action: {
                        engine.state = .menu
                    }) {
                        Text("BACK")
                            .font(.system(size: 16, weight: .bold))
                            .foregroundColor(.white)
                            .padding()
                    }
                }
                .background(Color.black.opacity(0.9))
            } else if engine.state == .playing {
                HStack(alignment: .top) {
                    // Home Button
                    Button(action: {
                        withAnimation {
                            engine.quitGame()
                        }
                    }) {
                        Image(systemName: "house.fill")
                            .font(.system(size: 20))
                            .foregroundColor(.white.opacity(0.6))
                            .padding(8)
                            .background(Color.black.opacity(0.3))
                            .clipShape(Circle())
                    }
                    .padding(.top, 40)
                    .padding(.leading, 16)
                    
                    VStack(alignment: .leading) {
                        Text(engine.levels[engine.currentLevelIndex].name)
                            .font(.caption)
                            .foregroundColor(.gray)
                        Text("\(engine.score / 10)")
                            .font(.system(size: 24, weight: .bold))
                            .foregroundColor(.white)
                    }
                    .padding(.top, 40) // Status bar area
                    .padding(.leading, 8)
                    
                    Spacer()
                    
                    if let target = engine.levels[engine.currentLevelIndex].targetScore {
                        VStack(alignment: .trailing) {
                            Text("Target")
                                .font(.caption)
                                .foregroundColor(.gray)
                            Text("\(target / 10)")
                                .font(.system(size: 24, weight: .bold))
                                .foregroundColor(.white.opacity(0.5))
                        }
                        .padding()
                        .padding(.top, 40)
                    }
                    if engine.scoreMultiplier > 1 {
                        Text("2x SCORE!")
                            .font(.system(size: 20, weight: .bold))
                            .foregroundColor(.yellow)
                            .padding(.top, 80)
                            .transition(.scale)
                    }
                    
                    if engine.emergencySpinsAvailable > 0 {
                        // Removed
                    }
                }
                Spacer()
            } else if engine.state == .levelComplete {
                Color.black.opacity(0.8)
                    .edgesIgnoringSafeArea(.all)
                    .overlay(
                        VStack {
                            Text("LEVEL COMPLETE")
                                .font(.system(size: 36, weight: .bold))
                                .foregroundColor(.yellow)
                            Text("Score: \(engine.score / 10)")
                                .font(.system(size: 20))
                                .foregroundColor(.white)
                                .padding(.top, 10)
                            
                            Text("Tap for Next Level")
                                .font(.system(size: 16))
                                .foregroundColor(.white)
                                .padding(.top, 40)
                                .opacity(engine.gameOverTimer % 60 < 30 ? 1 : 0.5) // Blink effect
                        }
                    )
                    .contentShape(Rectangle())
                    .onTapGesture {
                        engine.handleInput(left: true, right: false)
                    }
            } else if engine.state == .gameover {
                Color.black.opacity(0.8)
                    .edgesIgnoringSafeArea(.all)
                    .overlay(
                        VStack {
                            Text("GAME OVER")
                                .font(.system(size: 40, weight: .bold))
                                .foregroundColor(.white)
                            Text("Score: \(engine.score / 10)")
                                .font(.system(size: 20))
                                .foregroundColor(.white)
                                .padding(.top, 10)
                            if engine.highScore > 0 {
                                Text("Best: \(engine.highScore / 10)")
                                    .font(.system(size: 16))
                                    .foregroundColor(.yellow)
                                    .padding(.top, 10)
                            }
                            Text("Tap to Restart")
                                .font(.system(size: 16))
                                .foregroundColor(engine.gameOverTimer > 10 ? .white : .gray)
                                .padding(.top, 40)
                        }
                    )
                    .contentShape(Rectangle())
                    .onTapGesture {
                        engine.handleInput(left: true, right: false)
                    }
            }
        }
    }
    
    var inputLayer: some View {
        HStack(spacing: 0) {
            // Left Zone
            Color.clear
                .contentShape(Rectangle())
                .gesture(
                    DragGesture(minimumDistance: 0)
                        .onChanged { _ in engine.handleInput(left: true, right: engine.inputRight) }
                        .onEnded { _ in engine.handleInput(left: false, right: engine.inputRight) }
                )
            
            // Right Zone
            Color.clear
                .contentShape(Rectangle())
                .gesture(
                    DragGesture(minimumDistance: 0)
                        .onChanged { _ in engine.handleInput(left: engine.inputLeft, right: true) }
                        .onEnded { _ in engine.handleInput(left: engine.inputLeft, right: false) }
                )
        }
    }
}

extension Color {
    init(hex: UInt, alpha: Double = 1) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xff) / 255,
            green: Double((hex >> 08) & 0xff) / 255,
            blue: Double((hex >> 00) & 0xff) / 255,
            opacity: alpha
        )
    }
}

File: android/app/src/main/java/com/trae/twotones/GameView.kt
package com.trae.twotones

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.core.tween
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.animation.scaleIn
import androidx.compose.animation.scaleOut
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.gestures.detectDragGestures
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.graphics.drawscope.rotate
import androidx.compose.ui.graphics.drawscope.translate
import androidx.compose.ui.input.pointer.pointerInput
import androidx.compose.ui.layout.onSizeChanged
import androidx.compose.ui.platform.LocalDensity
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.IntSize
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel

@Composable
fun GameView(engine: GameEngine = viewModel()) {
    val state by engine.state.collectAsState()
    val density = LocalDensity.current
    
    BoxWithConstraints(modifier = Modifier.fillMaxSize()) {
        val maxWidthPx = with(density) { maxWidth.toPx().toDouble() }
        val maxHeightPx = with(density) { maxHeight.toPx().toDouble() }
        
        LaunchedEffect(maxWidthPx, maxHeightPx) {
            engine.setDimensions(maxWidthPx, maxHeightPx)
        }

        // Background
        Box(modifier = Modifier.fillMaxSize().background(Color(0xFF1A1A1A)))

        // Game Render Loop (Canvas)
        if (state == GameState.PLAYING || state == GameState.GAMEOVER || state == GameState.LEVEL_COMPLETE) {
            GameCanvas(engine, Modifier.fillMaxSize())
        }

        // UI Overlay
        UiOverlay(engine)

        // Input Layer (Transparent)
        if (state == GameState.PLAYING) {
            InputLayer(engine)
        }
    }
}

@Composable
fun GameCanvas(engine: GameEngine, modifier: Modifier) {
    val obstacles by engine.obstacles.collectAsState()
    val player by engine.player.collectAsState()
    val state by engine.state.collectAsState()
    
    Canvas(modifier = modifier) {
        // Draw Obstacles
        obstacles.forEach { obs ->
            var obsColor = Color.White
            if (obs.type == ObstacleType.VARIABLE_SPEED) {
                obsColor = Color.Yellow
            }
            
            if (obs.type == ObstacleType.ROTATING || obs.type == ObstacleType.DIAGONAL) {
                rotate(degrees = Math.toDegrees(obs.rotation).toFloat(), pivot = Offset((obs.x + obs.width/2).toFloat(), (obs.y + obs.height/2).toFloat())) {
                    drawRect(
                        color = obsColor,
                        topLeft = Offset(obs.x.toFloat(), obs.y.toFloat()),
                        size = Size(obs.width.toFloat(), obs.height.toFloat())
                    )
                }
            } else if (obs.type == ObstacleType.CROSS) {
                 // Horizontal
                 drawRect(
                     color = obsColor,
                     topLeft = Offset(obs.x.toFloat(), (obs.y + obs.height/3).toFloat()),
                     size = Size(obs.width.toFloat(), (obs.height/3).toFloat())
                 )
                 // Vertical
                 drawRect(
                     color = obsColor,
                     topLeft = Offset((obs.x + obs.width/2 - 15).toFloat(), obs.y.toFloat()),
                     size = Size(30f, obs.height.toFloat())
                 )
            } else {
                drawRect(
                    color = obsColor,
                    topLeft = Offset(obs.x.toFloat(), obs.y.toFloat()),
                    size = Size(obs.width.toFloat(), obs.height.toFloat())
                )
            }
        }
        
        // Draw Player
        if (state == GameState.PLAYING || state == GameState.GAMEOVER) {
            val centerX = size.width / 2
            val centerY = size.height * 0.8f
            
            // Ring
            drawCircle(
                color = Color.White.copy(alpha = 0.1f),
                radius = player.radius.toFloat(),
                center = Offset(centerX, centerY),
                style = Stroke(width = 2f)
            )
            
            // Balls
            val x1 = centerX + kotlin.math.cos(player.angle) * player.radius
            val y1 = centerY + kotlin.math.sin(player.angle) * player.radius
            
            val x2 = centerX + kotlin.math.cos(player.angle + Math.PI) * player.radius
            val y2 = centerY + kotlin.math.sin(player.angle + Math.PI) * player.radius
            
            drawCircle(
                color = Color(0xFFFF3B30),
                radius = player.ballRadius.toFloat(),
                center = Offset(x1.toFloat(), y1.toFloat())
            )
            
            drawCircle(
                color = Color(0xFF007AFF),
                radius = player.ballRadius.toFloat(),
                center = Offset(x2.toFloat(), y2.toFloat())
            )
        }
    }
}

@Composable
fun InputLayer(engine: GameEngine) {
    var leftDown by remember { mutableStateOf(false) }
    var rightDown by remember { mutableStateOf(false) }
    
    LaunchedEffect(leftDown, rightDown) {
        engine.handleInput(leftDown, rightDown)
    }
    
    Row(modifier = Modifier.fillMaxSize()) {
        TouchBox(
            modifier = Modifier.weight(1f).fillMaxSize(),
            onDownChange = { leftDown = it }
        )
        TouchBox(
            modifier = Modifier.weight(1f).fillMaxSize(),
            onDownChange = { rightDown = it }
        )
    }
}

@Composable
fun TouchBox(modifier: Modifier, onDownChange: (Boolean) -> Unit) {
    Box(
        modifier = modifier.pointerInput(Unit) {
            awaitPointerEventScope {
                while (true) {
                    val event = awaitPointerEvent()
                    val isDown = event.changes.any { it.pressed }
                    onDownChange(isDown)
                }
            }
        }
    )
}

@Composable
fun UiOverlay(engine: GameEngine) {
    val state by engine.state.collectAsState()
    val score by engine.score.collectAsState()
    val currentLevelIndex by engine.currentLevelIndex.collectAsState()
    val levels = engine.levels
    val unlockedLevel by engine.unlockedLevel.collectAsState()
    val highScore by engine.highScore.collectAsState()
    val scoreMultiplier = engine.scoreMultiplier
    
    if (state == GameState.MENU) {
        Column(
            modifier = Modifier.fillMaxSize().padding(bottom = 40.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center
        ) {
            Spacer(Modifier.weight(1f))
            
            // Logo
            Box(modifier = Modifier
                .size(80.dp)
                .rotate(-45f)
            ) {
                Canvas(modifier = Modifier.fillMaxSize()) {
                    drawCircle(color = Color.White.copy(alpha = 0.2f), style = Stroke(width = 8f))
                    drawCircle(color = Color(0xFFFF3B30), radius = 20f, center = center + Offset(40f, 0f))
                    drawCircle(color = Color(0xFF007AFF), radius = 20f, center = center + Offset(-40f, 0f))
                }
            }
            
            Spacer(Modifier.height(20.dp))
            
            Text("TWO TONES", fontSize = 50.sp, fontWeight = FontWeight.Bold, color = Color.White)
            
            Spacer(Modifier.height(40.dp))
            
            Button(
                onClick = { engine.startLevel(0) },
                colors = ButtonDefaults.buttonColors(containerColor = Color.White),
                modifier = Modifier.width(200.dp).height(60.dp)
            ) {
                Text("PLAY", fontSize = 24.sp, fontWeight = FontWeight.Bold, color = Color.Black)
            }
            
            Spacer(Modifier.height(20.dp))
            
            Button(
                onClick = { engine.setStateLevelSelect() },
                colors = ButtonDefaults.buttonColors(containerColor = Color.Transparent),
                border = androidx.compose.foundation.BorderStroke(2.dp, Color.White),
                modifier = Modifier.width(200.dp).height(50.dp)
            ) {
                Text("LEVELS", fontSize = 20.sp, fontWeight = FontWeight.SemiBold, color = Color.White)
            }
            
            Spacer(Modifier.weight(1f))
        }
    } else if (state == GameState.LEVEL_SELECT) {
        Column(
            modifier = Modifier.fillMaxSize().background(Color.Black.copy(alpha = 0.9f)),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Text("SELECT LEVEL", fontSize = 30.sp, fontWeight = FontWeight.Bold, color = Color.White, modifier = Modifier.padding(top = 60.dp))
            
            LazyVerticalGrid(
                columns = GridCells.Adaptive(minSize = 100.dp),
                modifier = Modifier.weight(1f).padding(20.dp),
                verticalArrangement = Arrangement.spacedBy(20.dp),
                horizontalArrangement = Arrangement.spacedBy(20.dp)
            ) {
                items(levels) { level ->
                    val isLocked = level.id > unlockedLevel
                    Box(
                        modifier = Modifier
                            .height(100.dp)
                            .clip(RoundedCornerShape(15.dp))
                            .background(if (isLocked) Color.Gray.copy(alpha = 0.3f) else Color.White.copy(alpha = 0.2f))
                            .clickable(enabled = !isLocked) { engine.startLevel(level.id - 1) },
                        contentAlignment = Alignment.Center
                    ) {
                        if (isLocked) {
                            Icon(Icons.Default.Lock, contentDescription = "Locked", tint = Color.Gray)
                        } else {
                            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                                Text("${level.id}", fontSize = 40.sp, fontWeight = FontWeight.Bold, color = Color.White)
                                Text(level.name, fontSize = 12.sp, color = Color.White.copy(alpha = 0.8f))
                            }
                        }
                    }
                }
            }
            
            Button(
                onClick = { engine.setStateMenu() },
                colors = ButtonDefaults.buttonColors(containerColor = Color.Transparent),
                modifier = Modifier.padding(bottom = 40.dp)
            ) {
                Text("BACK", fontSize = 16.sp, fontWeight = FontWeight.Bold, color = Color.White)
            }
        }
    } else if (state == GameState.PLAYING) {
        // HUD
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 60.dp, start = 20.dp, end = 20.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.Top
        ) {
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp), verticalAlignment = Alignment.Top) {
                Box(
                    modifier = Modifier
                        .size(44.dp)
                        .clip(CircleShape)
                        .background(Color.White.copy(alpha = 0.1f))
                        .clickable { engine.quitGame() },
                    contentAlignment = Alignment.Center
                ) {
                    Icon(Icons.Default.Home, contentDescription = "Home", tint = Color.White.copy(alpha = 0.8f))
                }
                
                Column {
                    Text(levels[currentLevelIndex].name, fontSize = 14.sp, fontWeight = FontWeight.Medium, color = Color.Gray)
                    Text("${score / 10}", fontSize = 28.sp, fontWeight = FontWeight.Bold, color = Color.White)
                }
            }
            
            levels[currentLevelIndex].targetScore?.let { target ->
                Column(horizontalAlignment = Alignment.End) {
                    Text("Target", fontSize = 14.sp, fontWeight = FontWeight.Medium, color = Color.Gray)
                    Text("${target / 10}", fontSize = 28.sp, fontWeight = FontWeight.Bold, color = Color.White)
                }
            }
        }
        
        if (scoreMultiplier > 1) {
             Box(modifier = Modifier.fillMaxSize().padding(top = 120.dp), contentAlignment = Alignment.TopCenter) {
                 Text("2x SCORE!", fontSize = 20.sp, fontWeight = FontWeight.Bold, color = Color.Yellow)
             }
        }
    } else if (state == GameState.LEVEL_COMPLETE) {
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(Color.Black.copy(alpha = 0.8f))
                .clickable { engine.handleInput(true, false) }, // Tap to continue
            contentAlignment = Alignment.Center
        ) {
            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text("LEVEL COMPLETE", fontSize = 36.sp, fontWeight = FontWeight.Bold, color = Color.Yellow)
                Text("Score: ${score / 10}", fontSize = 20.sp, color = Color.White)
                Spacer(Modifier.height(40.dp))
                Text("Tap for Next Level", fontSize = 16.sp, color = Color.White)
            }
        }
    } else if (state == GameState.GAMEOVER) {
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(Color.Black.copy(alpha = 0.8f))
                .clickable { engine.handleInput(true, false) }, // Tap to restart
            contentAlignment = Alignment.Center
        ) {
            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text("GAME OVER", fontSize = 40.sp, fontWeight = FontWeight.Bold, color = Color.White)
                Text("Score: ${score / 10}", fontSize = 20.sp, color = Color.White)
                if (highScore > 0) {
                    Text("Best: ${highScore / 10}", fontSize = 16.sp, color = Color.Yellow)
                }
                Spacer(Modifier.height(40.dp))
                Text("Tap to Restart", fontSize = 16.sp, color = Color.White)
            }
        }
    }
}
